### PR TITLE
Update to new synchronized session API

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Implements the changes [done in core](https://github.com/Particular/NServiceBus/pull/6342) to register the Synchronized storage session in the DI container.
The adapters have been removed as the storage session is directly available from the container.

### CODE REVIEW

File: src/NServiceBus.RavenDB.AcceptanceTests/NServiceBus.RavenDB.AcceptanceTests.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net472;netcoreapp3.1;net6.0</TargetFrameworks>
+    <TargetFrameworks>net472;net6.0</TargetFrameworks>
```
Comment (lailabougria) [inline src/NServiceBus.RavenDB.AcceptanceTests/NServiceBus.RavenDB.AcceptanceTests.csproj:4]: Same comment here, the ci.yml needs to be adjusted

**Review by lailabougria — APPROVED** [review]

**Review by lailabougria — APPROVED** [review]

**Comment by SzymonPobiega** [comment]
@DavidBoike @bording we followed the .NET 6 release checklist on this. Assigning you as reviewers because you mentioned you would like to double check. We would like to merge it on Monday by EOD.

**Comment by DavidBoike** [comment]
In src/NServiceBus.RavenDB.ClusterWide.AcceptanceTests/NServiceBus.RavenDB.ClusterWide.AcceptanceTests.csproj the `<Compile Remove="\**\…` expressions aren't working. You can tell by going to the [new test summary that's being generated](https://github.com/Particular/NServiceBus.RavenDB/actions/runs/2312422978). Search for `When_requesting_message_to_be_forwarded` go to the 2nd one, and scroll up, you'll see that it's being executed in NServiceBus.RavenDB.ClusterWide.AcceptanceTests. Luckily it passes anyway, but if we fix that hopefully the Windows tests wouldn't need 37.5 minutes to complete.

**Comment by DavidBoike** [comment]
Meant to say in the last comment that the solution is to add `GeneratePathProperty="true"` to the AcceptanceTests PackageReference, and then use `$(PkgNServiceBus_AcceptanceTests_Sources)` [like this](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/blob/master/src/Compatibility.StorageTable.AcceptanceTests/Compatibility.StorageTable.AcceptanceTests.csproj)

I looked through the PR and the net6.0 stuff seems fine to me. I skimmed right on past the persistence stuff.

**Review by tmasternak — APPROVED** [review]

**Pull request merged**